newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js

```diff
@@ -260,11 +264,13 @@ export const HomePage = React.memo<Props>(
           const projectMetadata = await onCreateBlank({
             i18n,
             outputPath,
+            projectName: newProjectName,
           });
           if (!projectMetadata) return;
           const { project, storageProvider, fileMetadata } = projectMetadata;
           setPreCreationDialogOpen(false);
           setOutputPath(computeDefaultProjectPath());
+          setNewProjectName(generateName());
```

Comment:
Is that necessary if you already have the state value intialised with the same function ?

Reply:
It is to make sure that if the user creates a new project, closes it and then creates another one, there is a new project name suggestion instead of the last project name he entered

Reply:
Related question: is this necessary to then initialize the name in the useState?

Reply:
The homepage is never unmounted so I would say yes. But there's maybe something I don't see